test/python/pulse/test_pulse_lib.py

```diff
@@ -127,16 +145,21 @@ def test_gauss_square_samples(self):
         duration = 125
         sigma = 4
         amp = 0.5j
+        width = 100
         # formulaic
-        times = np.array(range(25), dtype=np.complex_)
-        times = times - (25 / 2) + 0.5
-        gauss = amp * np.exp(-(times / sigma)**2 / 2)
+        gauss_duration = duration - width
+        times = np.array(range(gauss_duration), dtype=np.complex_)
+        times = times - (gauss_duration / 2) + 0.5
+        gauss, _ = self._zero_ends_gaussian(times, amp, sigma, gauss_duration)
```

Comment:
Why do you need to call this private method? Is there a way around this?

Reply:
Well I could call the method from `continuous.py`, but it seemed the idea of these tests was to avoid using anything from `pulse_lib` (since that's exactly how samples are generated).

I could write this logic in each test, but I need it in multiple tests so I thought using the function would be better practice.

Reply:
I think its fine to call this function, or just make sure the ends of the pulse are zero.

Reply:
Fixed in https://github.com/Qiskit/qiskit-terra/pull/4640/commits/e2082984e0793392b7a0d257fd8ed6b6d0a4a20f